shpkpr/exceptions.py

```diff
@@ -37,6 +37,8 @@ def wrapper(*args, **kwargs):
                 if six.PY2:
                     six.reraise(exception_to_rewrap_with, ei.message, tb)
                 else:
-                    six.reraise(exception_to_rewrap_with, exception_to_rewrap_with(*ei.args), tb)
+                    ei_args = ei.args if ei.args else (ei.message,) if hasattr(ei, "message") else None
+                    ei_args = ei_args if ei_args is not None else (str(ei),)
```

Comment:
I agree, I did it this way precisely because `mccabe` told me the function was too complex (i had a proper nested `if` before). I'll have a think about it, maybe split it out to a wee helper function.